content/chainguard/chainguard-images/working-with-images/incert-custom-certs.md

```diff
@@ -0,0 +1,269 @@
+* A tool for creating a self-signed certificate. This guide highlights using `cfssl`, a public key infrastructure toolkit from CloudFlare, but alternatives like `openssl` could also be used for this purpose. Follow [the `cfssl` installation instructions](https://github.com/cloudflare/cfssl#installation) to set this up.
+    * Note that if you use `cfssl`, you will also need the `cfssljson` utility installed as well. 
+
+
+## Creating a self-signed certificate
+
+First, let's create a directory to hold your certificate infrastructure.
+
+```sh
+mkdir ~/incert-example/ && cd $_
+```
+
+In the new directory, create a certificate signing request (CSR) by running the following command.
+
+```sh
+cat > csr.json <<EOF
+{
+	"hosts": [
+    	"example.com",
+    	"www.example.com"
+	],
+	"CN": "www.example.com",
+	"key": {
+    	"algo": "rsa",
+    	"size": 2048
+	},
+	"names": [{
+    	"C": "US",
+    	"L": "San Francisco",
+    	"O": "Example Company, LLC",
+    	"OU": "Operations",
+    	"ST": "California"
+	}]
+}
+EOF
+```
+
+We want to create some certificates for `example.com` and `www.example.com`, so we include these here in a list for the CSR's `hosts` value. This means the certificates will only be valid for these domains. 
+
+Next, create your certificates by running the following `cfssl selfsign` command.
+
+```sh
+cfssl selfsign www.example.com csr.json | cfssljson -bare selfsigned
+```
+
+Here we include the hostname we specified previously (`www.example.com`) as well as the CSR file. We then pipe the command's output into a `cfssljson` command; this will process the `.json` files output by the `cfssl selfsign` command into the `.pem` files we need.
+
+This command will return a warning that self-signed certificates are insecure. This is the expected behavior for `cfssl`, and since we are only using these certificates to demonstrate how `incert` works there won't be any security concerns.
+
+```
+. . .
+
+*** WARNING ***
+
+Self-signed certificates are dangerous. Use this self-signed
+certificate at your own risk.
+
+It is strongly recommended that these certificates NOT be used
+in production.
+
+*** WARNING ***
+```
+
+Following that, if you check the contents of your working directory you will find the self-signed CSR, the key, and the certificate.
+
+```sh
+ls
+```
+```
+csr.json  selfsigned.csr  selfsigned-key.pem  selfsigned.pem
+```
+
+With these files in place you can move on to creating an Nginx container that uses these certificates to provide TLS.
+
+## Create an Nginx container that uses self-signed certificates for TLS
+
+Now that you've created the certificate infrastructure, you can create an Nginx container that uses them to provide TLS. Later on, we will attempt to reach this Nginx container with a `curl` container we built using `incert`, testing whether it can successfully insert the `selfsigned.pem` certificate into it.
```

Comment:
Maybe "testing that incert correctly installed the `selfsigned.pem` certificate into it"?